### DISCUSSION TITLE
Can somebody explain aedes.subscribe (topic, deliverfunc, callback)

### DISCUSSION THREAD

**Discussion by aqibamir** [moscajs/aedes#809] [Q&A]
I've tried to understand `deliverfunc` from documentation but I just can not wrap my head around it.

**Comment by robertsLando** [comment]
- `topic` is the topic to subscribe to
- `deliverFunction` is the function called when a message is received matching the sub topic. It has `packet, done` args, packet is the published packet received and done is the funcction to call once you finish processing the packet. The reason about `done` is to provide backpressure, in poor words means that until you don't call `done` aedes will stop processing other packets (stops the incoming stream).
- `cb` is called when subscription is correctly setup on broker, it's optional BTW, it uses mqemitter: https://github.com/mcollina/mqemitter#emitteron-topic-listener-callback
